onnxmltools/utils/float16_converter.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+from struct import unpack
+from ..convert.common.data_types import *
+from onnxmltools.utils import load_model, save_model
+
+
+def npfloat16_to_int(np_list):
+    """
+    Convert numpy float16 to python int.
+    :param np_list: numpy float16 list
+    :return int_list: python int list
+    """
+    return [int(bin(_.view('H'))[2:].zfill(16), 2) for _ in np_list]
+
+
+def float16_converter(onnx_file, new_onnx_file):
```

Comment:
Does this function only convert FLOAT to FLOAT16? Or it also converts DOUBLE to FLOAT16?

Reply:
If it only converts FLOAT to FLOAT16, we need to throw if a DOUBLE is found.

Reply:
I thought DOUBLE is a seperate data type to handle if we want to convert it. Same as complex64 or complex128. Do we need to consider them during the conversion from FLOAT to FLOAT16?

Reply:
ONNX potentially allows a mixture of numerical types in a single model. Assume a user calls this float compressor and her model contains both of DOUBLE and FLOAT. It's a bit confusing if she can still find DOUBLE in her compressed model. This comment is relatively minor so it is not blocking you from merging this PR. :)